src/main/java/com/github/mtsochacki/adventofcode/Day18.java

```diff
@@ -0,0 +1,183 @@
+            this.level = regularNumber.level;
+            this.value = regularNumber.value;
+        }
+
+        public void incrementLevel() {
+            this.level++;
+        }
+
+        public void decrementLevel() {
+            this.level--;
+        }
+
+        public void increaseValue(int value) {
+            this.value += value;
+        }
+    }
+
+    @Override
+    public String part1(List<String> input) {
+        Integer slugMagnitude = input.stream()
+                .map(this::transformIntoSnailfishNumber)
+                .reduce(this::reduceSnailfishNumber)
+                .map(this::calculateMagnitude)
+                .orElseThrow(() -> new RuntimeException("Error processing SnailFish"));
+        return String.valueOf(slugMagnitude);
+    }
+
+    @Override
+    public String part2(List<String> input) {
+        List<List<RegularNumber>> snailfishNumbers = input.stream()
+                .map(this::transformIntoSnailfishNumber)
+                .toList();
+        int max = findHighestMagnitude(snailfishNumbers);
+        return String.valueOf(max);
+    }
+
+    private int findHighestMagnitude(List<List<RegularNumber>> snailfishNumbers) {
+        int max = 0;
+        for (int i = 0; i < snailfishNumbers.size(); i++) {
+            for (int j = 0; j < snailfishNumbers.size(); j++) {
+                if (i == j) {
+                    continue;
+                }
+                List<RegularNumber> firstSlug = copySnailfishNumber(snailfishNumbers.get(i));
+                List<RegularNumber> copyFirstSlug = copySnailfishNumber(snailfishNumbers.get(i));
+                List<RegularNumber> secondSlug = copySnailfishNumber(snailfishNumbers.get(j));
+                List<RegularNumber> copySecondSlug = copySnailfishNumber(snailfishNumbers.get(j));
+                List<RegularNumber> firstResult = reduceSnailfishNumber(firstSlug, secondSlug);
+                List<RegularNumber> secondResult = reduceSnailfishNumber(copySecondSlug, copyFirstSlug);
+                int newMax = Math.max(calculateMagnitude(firstResult), calculateMagnitude(secondResult));
+                max = Math.max(newMax, max);
+            }
+        }
+        return max;
+    }
+
+    private List<RegularNumber> copySnailfishNumber(List<RegularNumber> snailfishNumber) {
+        List<RegularNumber> newSnailfishNumber = new ArrayList<>();
+        for (RegularNumber regularNumber : snailfishNumber) {
+            newSnailfishNumber.add(new RegularNumber(regularNumber));
+        }
+        return newSnailfishNumber;
+    }
+
+    private List<RegularNumber> transformIntoSnailfishNumber(String line) {
+        List<RegularNumber> snailfishNumber = new ArrayList<>();
+        int level = 0;
+        for (char character : line.toCharArray()) {
+            if (character == '[') {
+                level++;
+            } else if (character == ']') {
+                level--;
+            } else if (Character.isDigit(character)) {
```

Comment:
If we want to be pedantic, I'd mark it somewhere that this assumes the `line` represents a *reduced* snailfish number. Otherwise, the digit parsing would not be enough

Reply:
noted